kartothek/core/dataset.py

```diff
@@ -381,16 +381,69 @@ def get_indices_as_dataframe(
         elif columns == []:
             return pd.DataFrame(index=self.partitions)
 
-        dfs = []
         columns_to_scan = columns[:]
         if predicates:
             predicate_columns = columns_in_predicates(predicates)
             # Don't use set logic to preserve order
+            # TODO why do I need to preserve the order here?
             for col in predicate_columns:
                 if col not in columns_to_scan and col in self.indices:
                     columns_to_scan.append(col)
 
-        for col in columns_to_scan:
+            dfs = [
+                self._evaluate_conjunction(
+                    columns=columns_to_scan,
+                    predicates=[conjunction],
+                    date_as_object=date_as_object,
+                )
+                for conjunction in predicates
+            ]
+
+            df = pd.concat(dfs)
```

Comment:
does this work without `ignore_index=True`?

Reply:
Wouldn't `ignore_index=True` drop the index entirely? We would lose the mapping to the partitions, wouldn't we?